### PR TITLE
plugin: add audit plugin extension point (#9136)

### PR DESCRIPTION
**Pull request by lysu**
cherry-pick #9136 to 2.1

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/9954)

### CODE REVIEW

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by zz-jason** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by zz-jason** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by zz-jason** [comment]
@lysu please fix CI and resolve conflicts.

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by lysu** [comment]
PTAL @jackysp

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

File: executor/set.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package executor
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/plugin"
```
Comment (tiancaiamao) [inline executor/set.go]: ...

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ package core
 import (
 	"bytes"
 	"fmt"
+	"github.com/pingcap/tidb/sessionctx/stmtctx"
```
Comment (tiancaiamao) [inline planner/core/planbuilder.go]: ...

**Comment by tiancaiamao** [comment]
LGTM
Please address comment

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by lysu** [comment]
@tiancaiamao @jackysp  PTAL

**Pull request merged**